atc/db/migration/migrations/1618553965_add_worker_name_to_worker_resource_caches.up.sql

```diff
@@ -0,0 +1,12 @@
+ALTER TABLE worker_resource_caches
+    ADD COLUMN worker_name text REFERENCES workers(name) ON DELETE CASCADE;
+
+DROP INDEX worker_resource_caches_uniq;
+
+CREATE UNIQUE INDEX worker_resource_caches_uniq
```

Comment:
I think it might be more beneficial to create this unique index in the order of `worker_name, resource_cache_id, worker_base_resource_type_Id`. This is because we will be querying for worker resource caches using `worker_name` and `resource_cache_id`, and not `worker_base_resource_type_id` so having  `worker_base_resource_type_id` as the first column in the index will result in those queries not being able to use this index. Another reason is that the index being split on `worker_name` first will help narrow down the scope of the queries using this index because it will be scoping down the rows in the table to the specific worker we are looking for, and then try to look for the resource cache within those rows.